### PR TITLE
dnsdist: Use SO_BINDTODEVICE when available for newServer's source itf

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdist-lua.cc [hunk 1]
```diff
@@ -237,10 +238,14 @@ void setupLuaConfig(bool client)
 			        sourceAddr = ComboAddress(source.substr(0, pos));
 			        sourceItf = itfIdx;
 			      }
+#ifdef SO_BINDTODEVICE
+                              /* we need to retain CAP_NET_RAW to be able to set SO_BINDTODEVICE in the health checks */
+                              g_capabilitiesToRetain.insert("CAP_NET_RAW");
```
Comment (omoerbeek) [inline pdns/dnsdist-lua.cc]: Indenting wrong or a github render error?

File: pdns/dnsdist-tcp.cc [hunk 2]
```diff
@@ -79,6 +79,12 @@ static std::unique_ptr<Socket> setupTCPDownstream(shared_ptr<DownstreamState>& d
         if (ds->ipBindAddrNoPort) {
           SSetsockopt(result->getHandle(), SOL_IP, IP_BIND_ADDRESS_NO_PORT, 1);
         }
+#endif
+#ifdef SO_BINDTODEVICE
+        int res = setsockopt(result->getHandle(), SOL_SOCKET, SO_BINDTODEVICE, ds->sourceItfName.c_str(), ds->sourceItfName.length());
```
Comment (omoerbeek) [inline pdns/dnsdist-tcp.cc]: Small nit: I had to check what happens if the name is empty. You might consider doing an explicit `if (!ds->sourceItfName.empty())`...

Comment (rgacogne) [reply pdns/dnsdist-tcp.cc]: Oh yes, indeed!

File: pdns/dnsdist-lua.cc (hunk 1 above)
Comment (rgacogne) [reply pdns/dnsdist-lua.cc]: I'll check but this function has a mix of indenting styles so it's always a pain. The upcoming reformat will be very welcome here!

**Pull request merged**